sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -297,21 +300,25 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
       .andWhere("offerings.institutionLocation.id = :locationId", {
         locationId,
       })
-      .andWhere("offerings.offeringType = :offeringType", {
-        offeringType: OfferingTypes.Public,
+      .andWhere("offerings.offeringType IN (:...offeringTypes)", {
+        offeringTypes: offeringsFilter.offeringTypes,
       })
       .andWhere(
         "offerings.studyStartDate BETWEEN programYear.startDate AND programYear.endDate",
       );
-    if (offeringIntensity) {
+    if (offeringsFilter.offeringIntensity) {
       query.andWhere("offerings.offeringIntensity = :offeringIntensity", {
-        offeringIntensity,
+        offeringIntensity: offeringsFilter.offeringIntensity,
+      });
+    }
+    if (offeringsFilter.offeringStatus) {
+      query.andWhere("offerings.offeringStatus = :offeringStatus", {
+        offeringStatus: offeringsFilter.offeringStatus,
       });
     }
```

Comment:
Since there are a conditional queries for attributes of offeringsFilter is the offeringsFilter parameter not supposed to be optional ?

Reply:
The filter itself is not optional as it needs to go with default value(s) like offering type = public.

I will change the offeringType in offeringsFilter to be mandatory as it has default value.